unit-tests/src/test/java/org/eclipse/collections/impl/AbstractRichIterableTestCase.java

```diff
@@ -1363,6 +1364,14 @@ public void toSortedMapBy()
         Verify.assertListsEqual(Lists.mutable.with(3, 2, 1), map.keySet().toList());
     }
 
+    @Test
+    public void toBiMap()
+    {
+        RichIterable<Integer> integers = this.newWith(1, 2, 3);
+        MutableBiMap<String, String> biMap = integers.toBiMap(Object::toString, Object::toString);
+        Assert.assertEquals(UnifiedMap.newWithKeysValues("1", "1", "2", "2", "3", "3"), biMap);
```

Comment:
This is what I mean by all elements are mapped to unique keys and values. You should be able to change the key function, value function, or both functions to one that returns a duplicate element (like `ignored -> "constant"`) and see that the behavior is incorrect.

Reply:
Have you checked it here https://github.com/eclipse/eclipse-collections/blob/4e177372405a7956bcc6a49c6281e72ce0e229ae/unit-tests/src/test/java/org/eclipse/collections/impl/AbstractRichIterableTestCase.java#L1375-L1387

Reply:
These test two of the three scenarios I mentioned, and I think the one that's still missing is the important one.

Reply:
You mean test with duplicate keys? Added that. 👍

Reply:
Yes! And I see it uncovered the bug and you've fixed it. I hope that wasn't too frustrating but I think it's more rewarding to add tests until you can find the one that's red and get it to go to green. I'm going to add a few more suggestions but we're down to lower priority style stuff and the overall change is in good shape.

Reply:
I made the changes in my fork and created a PR from mine to yours @sendilkumarn 

https://github.com/sendilkumarn/eclipse-collections/pull/1

You don't have to actually merge it. If you like the changes you could squash it for example. I made the PR to make it visible.